src/message/message.service.ts

```diff
@@ -77,4 +77,20 @@ export class MessageService {
     const message = await this.findOne(messageId); // throws NotFoundException if not found
     await this.messageRepository.remove(message);
   }
+
+  async getContextForChat(chatId: number): Promise<string[]> {
+    // *get all messages for this chatId
+    const messages = await this.messageRepository.find({
+      where: { chat: { id: chatId } },
+      order: { createdAt: 'ASC' }, // *order by creation date
+    });
+
+    // *return an array of message content
+    return messages.map((message) => message.content);
+  }
+
+  async clearChatHistory(chatId: number): Promise<void> {
```

Comment:
Why did you add this function? First off, we don't want to delete chat history! Besides, if we want to do that, you can call the `delete` method in the `ChatService` to delete a chat, and that will cascade deleting the messages associated with it. But above all, why do you need this operation?

Reply:
This is just based on a mockup of Bloom's restart feature, because I realized that users may want to start a new conversation within an hour but don't want to be connected to the previous one, so I followed the hints in the welcome message and designed this restart feature. However, I did notice that you mentioned in Notion that "to ensure transparency and maintain the integrity of the educational process, students are not permitted to delete or modify their conversations with the AI", so this is just a test version and I will remove it now.